File: solution/0027_remove_element.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int top = -1;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val)
                nums[++top] = nums[i];
        }
        return (top + 1);
    }
};

File: neetcode250/arrays-hashing/0027-remove-element.py
class Solution:
    def removeElement(self, nums: list[int], val: int) -> int:
        i, j = 0, 0
        while i < len(nums):
            if nums[i] == val:
                i += 1
            else:
                nums[j] = nums[i]
                i += 1
                j += 1
        return j